src/chronify/datetime_range_generator.py

```diff
@@ -65,8 +70,95 @@ def iter_timestamps(self) -> Generator[datetime, None, None]:
                     ):
                         yield pd.Timestamp(cur)
 
-    def list_distinct_timestamps_from_dataframe(self, df: pd.DataFrame) -> list[datetime]:
-        return sorted(df[self._model.time_column].unique())
-
     def list_time_columns(self) -> list[str]:
         return self._model.list_time_columns()
+
+    def list_distinct_timestamps_from_dataframe(self, df: pd.DataFrame) -> list[datetime]:  # TODO
+        result = sorted(df[self._model.time_column].unique())
+        if not isinstance(result[0], datetime):
+            result = [pd.Timestamp(x) for x in result]
```

Comment:
Is this supposed to be `datetime`? Or is the discrepancy related to the TODO on line 76?